scripts/route_check.py

```diff
@@ -489,6 +493,45 @@ def filter_out_standalone_tunnel_routes(routes):
     return updated_routes
 
 
+def get_soc_ips(config_db):
+    mux_table = config_db.get_table('MUX_CABLE')
+    soc_ips = []
+    for _, mux_entry in mux_table.items():
+        if "cable_type" in mux_entry and "soc_ipv4" in mux_entry:
+            soc_ips.append(mux_entry["soc_ipv4"])
+
+    return soc_ips
+
+
+def filter_out_soc_ip_routes(routes):
+    """
+    Ignore ASIC only routes for SOC IPs
+
+    For active-active cables, we want the tunnel route for SOC IPs
+    to only be programmed to the ASIC and not to the kernel. This is to allow
+    gRPC connections coming from ycabled to always use the direct link (since this
+    will use the kernel routing table), but still provide connectivity to any external
+    traffic in case of a link issue (since this traffic will be forwarded by the ASIC).
+    """
+    config_db = swsscommon.ConfigDBConnector()
+    config_db.connect()
+
+    if not is_dualtor(config_db):
```

Comment:
Is it a concern that this skip is applied to both active-active and active-standby?

Reply:
`active-standby` wouldn't have a `soc_ipv4` entry?

Reply:
I can also change `get_soc_ips` to make sure the cable type is active-active